probe/endpoint/dns_snooper.go

```diff
@@ -250,15 +250,14 @@ func (s *DNSSnooper) processDNSMessage(dns *layers.DNS) {
 		domainQueried = question.Name
 		records       = append(dns.Answers, dns.Additionals...)
 		ips           = map[string]struct{}{}
-		alias         []byte
+		aliases       = [][]byte{}
 	)
 
-	// Traverse records for a CNAME first since the DNS RFCs don't seem to guarantee it
-	// appearing before its A-records
+	// Traverse all the CNAME records and the get the aliases. There are cases when the A record is for only one of the
+	// aliases. We traverse CNAME records first because there is no guarantee that the A records will be the first ones
 	for _, record := range records {
-		if record.Type == layers.DNSTypeCNAME && record.Class == layers.DNSClassIN && bytes.Equal(domainQueried, record.Name) {
-			alias = record.CNAME
-			break
+		if record.Type == layers.DNSTypeCNAME && record.Class == layers.DNSClassIN {
```

Comment:
Why don't we need the `bytes.Equal(domainQueried, record.Name)` here now?

Reply:
In my experiments I found that the record name for the `star.c10r.facebook.com.` was not `graph.facebook.com`. It looked like a "cascading" CNAME where the record name is the one of the previous CNAME.

I do have a follow up question though - do you see any problems with this approach? I don't see why we would get CNAME answers for a different hostname in the same message.

Reply:
In my experiments it seemed that the CNAME was not for the queried domain but for another name in that record (`star.c10r.facebook.com.` record name was other than "graph.facebook.com.".

Reply:
I don't know.  I'm not familiar with the code or the problem space.
Would you consider encoding two or three of the cases you saw as tests for this code, so at least we know in future what you were working off?